openmrs/omod/src/main/java/org/openmrs/projectbuendia/webservices/rest/PatientResource.java

```diff
@@ -369,38 +367,18 @@ private static void setLocation(Patient patient, String locationUuid) {
     }
 
     private SimpleObject searchInner(RequestContext requestContext) throws ResponseException {
-        List<Patient> patients = new ArrayList<>();
+        List<Patient> patients;
         String patientId = requestContext.getParameter("id");  // single patient by ID; overrides
-        // "q"
         if (patientId != null) {
             List<PatientIdentifierType> idTypes = new ArrayList<>();
             idTypes.add(DbUtil.getMsfIdentifierType());
             patients = patientService.getPatients(null, patientId, idTypes, true);
         } else {
-            String nameQuery = requestContext.getParameter("q");
-            patients = filterPatients(nameQuery, patientService.getAllPatients());
+            patients = patientService.getAllPatients();
```

Comment:
Yes, maintaining a branch in each repo with the same name sounds pretty reasonable.  (Try to name it something more specific than sync-overhaul if you can.)

The client repo is a submodule of the buendia repo; the intention there was to use the submodule pointer to track coordinated changes between client and server (i.e. when you check out the entire buendia repo recursively you should get a client and server that interoperate).  In practice this means you'd merge the client feature branch to dev first; then in the server feature branch, update the client submodule to point at the newly merged client; then merge the server feature branch to dev.